sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/http/HttpClientConfig.java

```diff
@@ -30,7 +30,7 @@ public HttpClientConfig withPoolSize(int maxPoolSize) {
         return this;
     }
 
-    public HttpClientConfig withHttpProxy(InetSocketAddress proxy) {
+    public HttpClientConfig withProxy(ProxyOptions proxy) {
```

Comment:
changed the method name here since it could be non http?

Reply:
Although this is internal so doesn't matter - with `withProxy` looks good.

Reply:
thanks